### PR TITLE
[Device map] nn.Parameter don't have children

### PR DESCRIPTION
**Pull request by patrickvonplaten**
This is needed to allow to use the "device_map" for models that have `nn.Parameter` at the top level, such as the safety checker of `diffusers`: https://github.com/huggingface/diffusers/pull/772/files#r990269773

### CODE REVIEW

**Review by sgugger — COMMENTED** [review]
What's a reproducer? I'm not sure how you get to a parameter here since we only apply the `children` method.

**Comment by patrickvonplaten** [comment]
> What's a reproducer? I'm not sure how you get to a parameter here since we only apply the `children` method.

Haha that was fast. Here a repro:

```
from diffusers.pipelines.stable_diffusion import StableDiffusionSafetyChecker

pipe = StableDiffusionSafetyChecker.from_pretrained('CompVis/stable-diffusion-safety-checker', device_map='auto')
```

**Comment by sgugger** [comment]
Which branch should I check? On main I just get an error telling me device_map auto is not supported yet

**Comment by patrickvonplaten** [comment]
Ah sorry just updated it above, can you try:

```python
from diffusers.pipelines.stable_diffusion import StableDiffusionSafetyChecker

StableDiffusionSafetyChecker._no_split_modules = ["CLIPEncoderLayer"]
pipe = StableDiffusionSafetyChecker.from_pretrained('CompVis/stable-diffusion-safety-checker', device_map='auto')
```

File: src/accelerate/utils/modeling.py [hunk 1]
```diff
@@ -258,7 +258,7 @@ def get_max_layer_size(
     modules_to_treat = modules.copy()
     while len(modules_to_treat) > 0:
         module_name, module = modules_to_treat.pop(0)
-        modules_children = list(module.named_children())
+        modules_children = list(module.named_children()) if not isinstance(module, torch.nn.Parameter) else []
```
Comment (sgugger) [inline src/accelerate/utils/modeling.py:261]:
```suggestion
        modules_children = list(module.named_children()) if isinstance(module, torch.nn.Module) else []
```

**Review by sgugger — APPROVED** [review]
Thanks for the reproducer! The function is indeed used with some parameters inside accelerate, so this is needed.

Just suggested another test.

**Comment by patrickvonplaten** [comment]
Awesome! Can I merge after one :heavy_check_mark: here or wait for another reviewer?

**Comment by sgugger** [comment]
Nope, you can go ahead and merge!

**Pull request merged**